### PR TITLE
Convert path.data to String setting instead of List

### PR DESCRIPTION
**Pull request by rjernst**
Since multiple data path support has been removed, the Setting no longer
needs to support multiple values. This commit converts the
PATH_DATA_SETTING to a String setting from List<String>.

relates #71205

### CODE REVIEW

File: benchmarks/src/main/java/org/elasticsearch/benchmark/fs/AvailableIndexFoldersBenchmark.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ public void setup() throws IOException {
         LogConfigurator.setNodeName("test");
         Settings settings = Settings.builder()
             .put(Environment.PATH_HOME_SETTING.getKey(), path)
-            .putList(Environment.PATH_DATA_SETTING.getKey(), paths)
+            .put(Environment.PATH_DATA_SETTING.getKey(), path.resolve("data"))
```
Comment (jasontedor) [inline benchmarks/src/main/java/org/elasticsearch/benchmark/fs/AvailableIndexFoldersBenchmark.java:53]: Is `paths` dead now?

**Review by jasontedor — APPROVED** [review]
LGTM. The core production change looks good. Skimmed the rest, if the compiler is happy, I’m happy. Left one comment.

File: server/src/main/java/org/elasticsearch/env/Environment.java [hunk 2]
```diff
@@ -179,7 +170,7 @@ public Settings settings() {
      * The data location.
      */
     public Path[] dataFiles() {
-        return dataFiles;
+        return new Path[] { dataFile };
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/env/Environment.java:173]: Could we (naively) inline this method right now, or in an imminent follow-up? That would leave us with a bunch of places where we do things like this:

```
        for (Path dataPath : new Path[]{environment.dataFile()}) {
```

I think this is preferable to using `environment.dataFiles()[0]` to get hold of the unique data path using the knowledge that it's always a singleton array. Unrolling loops like this is obviously correct (indeed IntelliJ will do it for you).

Comment (jasontedor) [reply server/src/main/java/org/elasticsearch/env/Environment.java:173]: Yeah, we discussed the approach here to not try to do MDP legacy code removal all at once, keeping each change relatively self-contained, and this is one we agreed to hold for a follow-up.

Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/env/Environment.java:173]: Acked. An imminent follow-up, or do we expect to merge things like #72278 first? We call this method in <30 places, so it's a pretty small change that IMO makes the subsequent larger cleanups much neater.

Comment (jasontedor) [reply server/src/main/java/org/elasticsearch/env/Environment.java:173]: Yeah, after this change, although we didn’t discuss a dependency on #72278. You’re making a compelling case there should be. Thank you!

Comment (rjernst) [reply server/src/main/java/org/elasticsearch/env/Environment.java:173]: I opened https://github.com/elastic/elasticsearch/pull/72327

**Pull request merged**